x-pack/plugins/ml/common/types/anomaly_detection_jobs/job.ts

```diff
@@ -33,3 +34,9 @@ export interface PerPartitionCategorization {
 }
 
 export type CustomSettings = estypes.MlCustomSettings;
+
+export function isAnomalyDetectionJob(arg: unknown): arg is Job {
+  return (
+    isPopulatedObject(arg) && typeof arg.job_id === 'string' && arg.analysis_config !== undefined
```

Comment:
Nit: You could move part of the check to `isPopulatedObject` here like this:

```
isPopulatedObject(arg, ['job_id', 'analysis_config']) && typeof arg.job_id === 'string'
```

Reply:
Updated in 7ee455c02c2252a159431305bf8b90cabaed6305